xray/core/plotting.py

```diff
@@ -0,0 +1,295 @@
+            plotfunc = plot_imshow
+        else:
+            plotfunc = plot_contourf
+    else:
+        plotfunc = plot_hist
+
+    kwargs['ax'] = ax
+    return plotfunc(darray, **kwargs)
+
+
+# This function signature should not change so that it can use
+# matplotlib format strings
+def plot_line(darray, *args, **kwargs):
+    """
+    Line plot of 1 dimensional DataArray index against values
+
+    Wraps matplotlib.pyplot.plot
+
+    Parameters
+    ----------
+    darray : DataArray
+        Must be 1 dimensional
+    ax : matplotlib axes object
+        If not passed, uses the current axis
+    args, kwargs
+        Additional arguments to matplotlib.pyplot.plot
+
+    """
+    import matplotlib.pyplot as plt
+
+    ndims = len(darray.dims)
+    if ndims != 1:
+        raise ValueError('Line plots are for 1 dimensional DataArrays. '
+                         'Passed DataArray has {} dimensions'.format(ndims))
+
+    # Ensures consistency with .plot method
+    ax = kwargs.pop('ax', None)
+
+    if ax is None:
+        ax = plt.gca()
+
+    xlabel, x = list(darray.indexes.items())[0]
+
+    _ensure_plottable([x])
+
+    primitive = ax.plot(x, darray, *args, **kwargs)
+
+    ax.set_xlabel(xlabel)
+
+    if darray.name is not None:
+        ax.set_ylabel(darray.name)
+
+    return primitive
+
+
+def plot_hist(darray, ax=None, **kwargs):
+    """
+    Histogram of DataArray
+
+    Wraps matplotlib.pyplot.hist
+
+    Plots N dimensional arrays by first flattening the array.
+
+    Parameters
+    ----------
+    darray : DataArray
+        Can be any dimension
+    ax : matplotlib axes object
+        If not passed, uses the current axis
+    kwargs :
+        Additional keyword arguments to matplotlib.pyplot.hist
+
+    """
+    import matplotlib.pyplot as plt
+
+    if ax is None:
+        ax = plt.gca()
+
+    no_nan = np.ravel(darray)
+    no_nan = no_nan[np.logical_not(np.isnan(no_nan))]
+
+    primitive = ax.hist(no_nan, **kwargs)
+
+    ax.set_ylabel('Count')
+
+    if darray.name is not None:
+        ax.set_title('Histogram of {}'.format(darray.name))
+
+    return primitive
+
+
+def _update_axes_limits(ax, xincrease, yincrease):
+    """
+    Update axes in place to increase or decrease
+    For use in _plot2d
+    """
+    if xincrease is None:
+        pass
+    elif xincrease:
+        ax.set_xlim(sorted(ax.get_xlim()))
+    elif not xincrease:
+        ax.set_xlim(sorted(ax.get_xlim(), reverse=True))
+
+    if yincrease is None:
+        pass
+    elif yincrease:
+        ax.set_ylim(sorted(ax.get_ylim()))
+    elif not yincrease:
+        ax.set_ylim(sorted(ax.get_ylim(), reverse=True))
+
+
+def _plot2d(plotfunc):
+    """
+    Decorator for common 2d plotting logic.
+    """
+    commondoc = '''
+    Parameters
+    ----------
+    darray : DataArray
+        Must be 2 dimensional
+    ax : matplotlib axes object
+        If None, uses the current axis
+    xincrease : None (default), True, or False
```

Comment:
Usually I prefer not to state defaults in the docstrings when they are also in the function signatures.